### PR TITLE
Validator for item quantity type; unit tests for request creation.

### LINKED ISSUE #164: [BACKEND] Implement validator for item quantity type field.

### LINKED ISSUE #173: [BACKEND] Create unit tests for request creation API.

### PR DESCRIPTION
**Pull request by git-adithyanair**

### CODE REVIEW

File: backend/api/community_get.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 type GetCommunityRequest struct {
-	ID int64 `uri:"id" binding:"min=0"`
+	ID int64 `uri:"id" binding:"required,min=0"`
```
Comment (yililiu8) [inline backend/api/community_get.go]:
```suggestion
        ID int64 `uri:"id" binding:"required,min=1"`
```
small nit but technically id's cant be zero right?

File: backend/api/request_create.go [hunk 2]
```diff
@@ -14,15 +14,15 @@ import (
...
-	QuantityType   db.ItemQuantityType `json:"quantity_type" binding:"required,min=2"`
+	QuantityType   db.ItemQuantityType `json:"quantity_type" binding:"required,min=2,quantity_type"`
 	Quantity       float64             `json:"quantity" binding:"required,min=1"`
 	PreferredBrand string              `json:"preferred_brand"`
 	Image          string              `json:"image"`
 	ExtraNotes     string              `json:"extra_notes"`
 }
 
 type CreateRequestRequest struct {
-	CommunityID int64                      `json:"community_id" binding:"required,min=1"`
```
Comment (yililiu8) [inline backend/api/request_create.go]: any reason why the min was deleted?

Comment (git-adithyanair) [reply backend/api/request_create.go]: I think I was just dumb

File: backend/api/request_create.go [hunk 3]
```diff
@@ -14,15 +14,15 @@ import (
...
+	CommunityID int64                      `json:"community_id" binding:"required"`
```
Comment (yililiu8) [inline backend/api/request_create.go]:
```suggestion
	CommunityID int64                      `json:"community_id" binding:"required,min=1"`
```

**Pull request merged**